packages/core/src/storage/compare.ts

```diff
@@ -362,7 +362,15 @@ export class StorageLayoutComparator {
       }
 
       case 't_userDefinedValueType': {
-        if (original.item.numberOfBytes === undefined || updated.item.numberOfBytes === undefined) {
+        const underlyingMatch =
+          original.item.underlying &&
+          updated.item.underlying &&
```

Comment:
It would be clearer to specifically check that these are not `undefined`:
```suggestion
          original.item.underlying !== undefined &&
          updated.item.underlying !== undefined &&
```